### PR TITLE
gstreamer: Add AudioLoader

### PR DESCRIPTION
**Pull request by KitaitiMakoto**
Hi,

I added `Gst::AudioLoader` to use [`GstAudioInfo`](https://gstreamer.freedesktop.org/documentation/audio/audio-info.html?gi-language=c) for #1634

Thank you.

### CODE REVIEW

File: gstreamer/test/test-audio.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class TestAudio < Test::Unit::TestCase
+  def test_audio
+    caps = Gst::Caps.new("audio/ogg")
+    caps.set_value("format", "S8")
+    caps.set_value("rate", 44100)
+    caps.set_value("channels", 2)
+
+    success, audio_info = Gst::AudioInfo.from_caps(caps)
+    assert(success)
```
Comment (kou) [inline gstreamer/test/test-audio.rb:25]: Hmm. Does `Gst::AudioInfo.new(caps)` work?

File: gstreamer/test/test-audio.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    structure = audio_info.to_caps.structures[0]
+    assert_equal(44100, structure.get_value("rate").value)
+    assert_equal(2, structure.get_value("channels").value)
+
+    audio_info.set_format(Gst::AudioFormat::F32LE, 16000, 1)
```
Comment (kou) [inline gstreamer/test/test-audio.rb:30]:
```suggestion
    audio_info.set_format(:f32le, 16000, 1)
```

File: gstreamer/test/test-audio.rb (hunk 1 above)
Comment (KitaitiMakoto) [reply gstreamer/test/test-audio.rb]: @kou 
Yes. I committed and pushed.

How did you know that? [`gst_audio_info_new`](https://gstreamer.freedesktop.org/documentation/audio/audio-info.html?gi-language=c#gst_audio_info_new) signature doesn't indicate any argument. I want to know it to infer GObject Introspection behavior.

Comment (kou) [reply gstreamer/test/test-audio.rb]: gobject-introspection gem creates a constructor (`#initialize`) that dispatches underlying constructors (`gst_audio_info_new()` and `gst_audio_info_new_from_caps()` in this case) automatically.

In this case, we can distinct `gst_audio_info_new()` and `gst_audio_info_new_caps()` calls by the given arguments. So `Gst::AudioInfo.new(caps)` works.

**Comment by kou** [comment]
It seems that `gst_audio_info_new_caps()` doesn't exist in old GStreamer:

https://github.com/ruby-gnome/ruby-gnome/actions/runs/12392635285/job/34592224364?pr=1641#step:11:2676

```text
   ===============================================================================
  Error: test_audio(TestAudio):
    ArgumentError: wrong arguments: GstAudio::AudioInfo#initialize(#<Gst::Caps:0x000055b44eae30f0 ptr=0x000055b44e5b9ad0 own=true audio/ogg, format=(string)S8, rate=(int)44100, channels=(int)2>):
    available signatures:
    GstAudio::AudioInfo#initialize()
  /ruby-gnome/gobject-introspection/lib/gobject-introspection/loader.rb:380:in `initialize'
  /ruby-gnome/gstreamer/test/test-audio.rb:24:in `new'
  /ruby-gnome/gstreamer/test/test-audio.rb:24:in `test_audio'
       21:     caps.set_value("rate", 44100)
       22:     caps.set_value("channels", 2)
       23: 
    => 24:     audio_info = Gst::AudioInfo.new(caps)
       25:     structure = audio_info.to_caps.structures[0]
       26:     assert_equal(44100, structure.get_value("rate").value)
       27:     assert_equal(2, structure.get_value("channels").value)
  ===============================================================================
```

Could you add `only_gstreamer_version(MAJOR, MINOR, MICRO)` in the added test?

File: gstreamer/test/test-audio.rb (hunk 1 above)
Comment (KitaitiMakoto) [reply gstreamer/test/test-audio.rb]: How flexible! Thank you for describing.

**Comment by KitaitiMakoto** [comment]
Done!

**Comment by kou** [comment]
Thanks.

**Pull request merged**